cw_bitcoin/lib/litecoin_wallet_addresses.dart

```diff
@@ -15,15 +24,85 @@ abstract class LitecoinWalletAddressesBase extends ElectrumWalletAddresses with
     required super.mainHd,
     required super.sideHd,
     required super.network,
+    required this.mwebHd,
+    required this.mwebEnabled,
     super.initialAddresses,
     super.initialRegularAddressIndex,
     super.initialChangeAddressIndex,
-  }) : super(walletInfo);
+  }) : super(walletInfo) {
+    topUpMweb(0);
+  }
+
+  final Bip32Slip10Secp256k1 mwebHd;
+  bool mwebEnabled;
+
+  List<int> get scanSecret => mwebHd.childKey(Bip32KeyIndex(0x80000000)).privateKey.privKey.raw;
+  List<int> get spendPubkey =>
+      mwebHd.childKey(Bip32KeyIndex(0x80000001)).publicKey.pubKey.compressed;
+
+  List<String> mwebAddrs = [];
+
+  Future<void> topUpMweb(int index) async {
+    // generate up to index + 1000 addresses:
+    while (mwebAddrs.length - index < 1000) {
+      final length = mwebAddrs.length;
+      final address = await CwMweb.address(
+        Uint8List.fromList(scanSecret),
+        Uint8List.fromList(spendPubkey),
+        length,
+      );
+      mwebAddrs.add(address!);
+    }
+  }
 
   @override
-  String getAddress(
-          {required int index,
-          required Bip32Slip10Secp256k1 hd,
-          BitcoinAddressType? addressType}) =>
-      generateP2WPKHAddress(hd: hd, index: index, network: network);
+  String getAddress({
+    required int index,
+    required Bip32Slip10Secp256k1 hd,
+    BitcoinAddressType? addressType,
+  }) {
+    if (addressType == SegwitAddresType.mweb) {
+      topUpMweb(index).then((value) {
+        return hd == sideHd ? mwebAddrs[0] : mwebAddrs[index + 1];
+      });
+    }
+    return generateP2WPKHAddress(hd: hd, index: index, network: network);
```

Comment:
this will cause the normal litcoin address to be returned, we already discussed testing our changes before pushing them, these types of issues should be easily discovered when simple debugging

Reply:
right so then we can't get rid of getAddressAsync() since topUpMweb is still asynchronous